### PR TITLE
FEAT-#6830: Remove public s3 bucket reference

### LINKED ISSUE #6830: Remove access to public s3 buckets
**Is your feature request related to a problem? Please describe.**
This is a compliance issue which comes from a principle of least privilege.

### PR DESCRIPTION
**Pull request by leshikus**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #6830 
- [ ] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

**Comment by leshikus** [comment]
`flake8` and `black` report many issues, none of them produced by my PR

**Comment by anmyachev** [comment]
> `flake8` and `black` report many issues, none of them produced by my PR

In CI or locally? If locally, check the versions of these packages. It should be like this: `black>=23.1.0` and `flake8>=6.0.0`.

**Comment by leshikus** [comment]
@anmyachev thanks, installing `pip` versions worked

**Comment by anmyachev** [comment]
@leshikus searching for `modin-datasets` gives two more files that use the S3 bucket: `test_io.py` and `test_io_exp.py`. Should they also be changed?

**Comment by leshikus** [comment]
@anmyachev thanks, a good catch; I believe I better create another PR for these tests because the change is not that easy.
I need either to replace S3 filesystem with HTTPS access or add credentials for that access. I wonder which approach is better.

File: examples/jupyter/Modin_Taxi.ipynb [hunk 1]
```diff
@@ -13,7 +13,7 @@
    "source": [
     "# To run this notebook as done in the README GIFs, you must first locally download the 2015 NYC Taxi Trip Data.\n",
     "import urllib.request\n",
-    "url_path = \"https://modin-datasets.s3.amazonaws.com/testing/yellow_tripdata_2015-01.csv\"\n",
+    "url_path = \"https://modin-datasets.intel.com/testing/yellow_tripdata_2015-01.csv\"\n",
```
Comment (anmyachev) [inline examples/jupyter/Modin_Taxi.ipynb:16]: @leshikus Modin users will no longer be able to run these examples locally? (because of credentials)

Comment (leshikus) [reply examples/jupyter/Modin_Taxi.ipynb:16]: HTTPS access will work without credentials

**Comment by anmyachev** [comment]
> @anmyachev thanks, a good catch; I believe I better create another PR for these tests because the change is not that easy.

ok.

>  I need either to replace S3 filesystem with HTTPS access or add credentials for that access. I wonder which approach is better.

I think that we need to leave s3 urls, since there may be a specific implementation for them, different from https. Maybe we should just use a local s3 server instead of the one in AWS.

**Review by anmyachev — DISMISSED** [review]
LGTM!

@YarShev any objections?

File: docs/getting_started/quickstart.rst [hunk 2]
```diff
@@ -64,13 +64,13 @@ For the purpose of demonstration, we will load in modin as ``pd`` and pandas as
...
 
-In this toy example, we look at the NYC taxi dataset, which is around 200MB in size. You can download `this dataset <https://modin-datasets.s3.amazonaws.com/testing/yellow_tripdata_2015-01.csv>`_ to run the example locally.
+In this toy example, we look at the NYC taxi dataset, which is around 200MB in size. You can download `this dataset <https://modin-datasets.intel.com/testing/yellow_tripdata_2015-01.csv>`_ to run the example locally.
 
 .. code-block:: python
 
   # This may take a few minutes to download
   import urllib.request
-  s3_path = "https://modin-datasets.s3.amazonaws.com/testing/yellow_tripdata_2015-01.csv"
+  s3_path = "https://modin-datasets.intel.com/testing/yellow_tripdata_2015-01.csv"
```
Comment (YarShev) [inline docs/getting_started/quickstart.rst:73]:
```suggestion
  url_path = "https://modin-datasets.intel.com/testing/yellow_tripdata_2015-01.csv"
```
Please replace this var everywhere in this file.

File: examples/quickstart.ipynb [hunk 3]
```diff
@@ -105,7 +105,7 @@
    "source": [
     "# This may take a few minutes to download\n",
     "import urllib.request\n",
-    "s3_path = \"https://modin-datasets.s3.amazonaws.com/testing/yellow_tripdata_2015-01.csv\"\n",
+    "s3_path = \"https://modin-datasets.intel.com/testing/yellow_tripdata_2015-01.csv\"\n",
```
Comment (YarShev) [inline examples/quickstart.ipynb:108]:
```suggestion
    "url_path = \"https://modin-datasets.intel.com/testing/yellow_tripdata_2015-01.csv\"\n",
```
Please replace this var everywhere in this file.

**Comment by YarShev** [comment]
> > I need either to replace S3 filesystem with HTTPS access or add credentials for that access. I wonder which approach is better.
> 
> I think that we need to leave S3 URLs, since there may be a specific implementation for them, different from HTTPS. Maybe we should just use a local S3 server instead of the one in AWS.

@anmyachev, should we add additional tests from reading files from the new location?

File: examples/quickstart.ipynb (hunk 3 above)
Comment (leshikus) [reply examples/quickstart.ipynb:108]: `url` is actually a path already, I'd better use `dataset_url`

Comment (leshikus) [reply examples/quickstart.ipynb]: Fixed

**Review by YarShev — APPROVED** [review]

**Review by anmyachev — APPROVED** [review]

**Comment by anmyachev** [comment]
> > > I need either to replace S3 filesystem with HTTPS access or add credentials for that access. I wonder which approach is better.
> > 
> > 
> > I think that we need to leave S3 URLs, since there may be a specific implementation for them, different from HTTPS. Maybe we should just use a local S3 server instead of the one in AWS.
> 
> @anmyachev, should we add additional tests for reading files from the new location?

What do you mean by new location? I believe it may potentially be useful to add tests for different protocols (https, s3)

**Comment by leshikus** [comment]
@anmyachev it seems the tests for notebooks are already in place, please disregard my comment about location

**Comment by anmyachev** [comment]
@leshikus please sign your commits with `-s` option also (not only `-S`). After that I will merge the PR.

**Review by anmyachev — APPROVED** [review]

**Comment by anmyachev** [comment]
Thanks @leshikus!

**Comment by YarShev** [comment]
> I believe it may potentially be useful to add tests for different protocols (https, s3)

Agree

**Pull request merged**